cypress/e2e/ui/Settings/Application-Settings/tenant.cy.js

```diff
@@ -0,0 +1,628 @@
+  accessControlAccordionItemId,
+  manageIQRegionAccordItem,
+  tenantsAccordionItem,
+
+  // Field values
+  formHeaderFragment,
+  initialParentTenantName,
+  initialParentTenantDescription,
+  editedTenantNameValue,
+  editedDescriptionValue,
+  projectNameValue,
+  initialChildTenantName,
+  initialChildTenantDescription,
+
+  // Element ids
+  nameInputFieldId,
+  descriptionInputFieldId,
+
+  // Config options
+  configToolbarButton,
+  editTenantConfigOption,
+  addProjectConfigOption,
+  deleteItemConfigOption,
+  manageQuotasConfigOption,
+  addChildTenantConfigOption,
+
+  // Quota Descriptions
+  allocatedStorageQuota,
+  allocatedVmQuota,
+
+  // Buttons
+  saveButton,
+  cancelButton,
+  resetButton,
+  addButton,
+
+  // Flash message text snippets
+  flashMessageOperationCanceled,
+  flashMessageOperationReset,
+  flashMessageSaved,
+  flashMessageAdded,
+  flashMessageCantDelete,
+  browserAlertDeleteConfirmText,
+  nameAlreadyTakenError,
+} = textConstants;
+
+function confirmUiNavigation(callback) {
+  cy.url()
+    ?.then((url) => {
+      // Ensures navigation to Settings -> Application-Settings in the UI
+      if (!url?.includes(componentRouteUrl)) {
+        // Navigate to Settings -> Application-Settings before looking out for parent Tenants edited during test
+        cy.menu(settingsMenuOption, appSettingsMenuOption);
+      }
+    })
+    .then(() => {
+      callback();
+    });
+}
+
+function cancelForm(
+  assertFlashMessage = true,
+  flashMessageSnippet = flashMessageOperationCanceled
+) {
+  cy.getFormFooterButtonByType(cancelButton).click();
+  if (assertFlashMessage) {
+    cy.expect_flash(flashClassMap.warning, flashMessageSnippet);
+  }
+}
+
+function resetForm(buttonType = 'reset') {
+  cy.getFormFooterButtonByType(resetButton, buttonType)
+    .should('be.enabled')
+    .click();
+  cy.expect_flash(flashClassMap.warning, flashMessageOperationReset);
+}
+
+function saveForm({
+  assertFlashMessage = true,
+  button = saveButton,
+  flashMessageSnippet = flashMessageSaved,
+} = {}) {
+  cy.getFormFooterButtonByType(button, 'submit').should('be.enabled').click();
+  if (assertFlashMessage) {
+    cy.expect_flash(flashClassMap.success, flashMessageSnippet);
+  }
+}
+
+function updateNameAndDescription(nameValue, descriptionValue) {
+  cy.getFormInputFieldById(nameInputFieldId).clear().type(nameValue);
+  cy.getFormInputFieldById(descriptionInputFieldId)
+    .clear()
+    .type(descriptionValue);
+}
+
+function verifyNameAndDescriptionValues(nameValue, descriptionValue) {
+  cy.getFormInputFieldById(nameInputFieldId).should('have.value', nameValue);
+  cy.getFormInputFieldById(descriptionInputFieldId).should(
+    'have.value',
+    descriptionValue
+  );
+}
+
+function validateFormElements(isEditForm = true) {
+  // Assert form header is visible
+  cy.expect_explorer_title(formHeaderFragment);
+  // Assert name field label is visible
+  cy.getFormLabelByInputId(nameInputFieldId).should('be.visible');
```

Comment:
Do we care about the `getFormLabelByInputId`?  We only check that the element is in the dom and has that id.  We don't check the value elsewhere.  If we care, we should check the id has a value or remove it entirely.  Thoughts?